qa/pull-tester/rpc-tests.py

```diff
@@ -266,7 +268,16 @@ def main():
         subprocess.check_call((config["environment"]["SRCDIR"] + '/qa/rpc-tests/' + test_list[0]).split() + ['-h'])
         sys.exit(0)
 
-    run_tests(test_list, config["environment"]["SRCDIR"], config["environment"]["BUILDDIR"], config["environment"]["EXEEXT"], args.jobs, args.coverage, passon_args)
+    if (args.rpcgroup == -1) != (args.machines == -1):
+        print("ERROR: Please use both -m and -r options when using parallel rpc_groups.")
+        sys.exit(0)
+    if args.rpcgroup != -1 and args.machines != -1 and args.rpcgroup < args.machines:
+        k, m = divmod(len(test_list), args.machines)
+        k = k if m == 0 else k+1
+        split_list = list(test_list[i*k:min((i+1)*k, len(test_list))] for i in range(args.machines))
+        run_tests(split_list[args.rpcgroup], config["environment"]["SRCDIR"], config["environment"]["BUILDDIR"], config["environment"]["EXEEXT"], args.jobs, args.coverage, passon_args)
+    else:
+        run_tests(test_list, config["environment"]["SRCDIR"], config["environment"]["BUILDDIR"], config["environment"]["EXEEXT"], args.jobs, args.coverage, passon_args)
```

Comment:
This would be clearer with a temporary:
```suggestion
        tests_to_run = split_list[args.rpcgroup]
    else:
        tests_to_run = test_list
    run_tests(tests_to_run, config["environment"]["SRCDIR"], config["environment"]["BUILDDIR"], config["environment"]["EXEEXT"], args.jobs, args.coverage, passon_args)
```